### DISCUSSION TITLE
Support Clone on error types

### DISCUSSION THREAD

**Discussion by Bajix** [diesel-rs/diesel#3262] [Ideas]
For batching queries it is necessary to be able to clone errors across all receivers. Perhaps we could add support for this by using a library that supports string cloning, such as [flexstr](https://crates.io/crates/flexstr) or [arcstr](https://crates.io/crates/arcstr)

**Comment by weiznich** [comment]
I fear that's not really possible because diesel uses boxed errors in quite a few places (`Box<dyn std::error::Error>`). We do not even control the inner error type in some cases. Introducing an additional `Clone` bound there would be a huge breaking change. Additionally it seems like clonable error types is generally nothing that is often used in the rust ecosystem.
